ForumProject/users/views.py

```diff
@@ -1,10 +1,143 @@
+        """
+        Handle GET requests to verify email activation.
+
+        Parameters:
+        - request: The request object.
+        - token: The activation token extracted from the URL.
+
+        Returns:
+        - A Response object with the result of the email verification.
+            - If the token is valid and the associated user is successfully activated,
+            returns a success message with HTTP status code 200.
+            - If the token is expired, returns an error message indicating the activation has expired
+            with HTTP status code 400.
+            - If the token is invalid, returns an error message indicating an invalid token
+            with HTTP status code 400.
+        """
+        try:
+            payload = jwt.decode(token, settings.SECRET_KEY, settings.SIMPLE_JWT['ALGORITHM'])
+            user = CustomUser.objects.get(id=payload['user_id'])
+            if not user.is_active:
+                user.is_active = True
+                user.save()
+            return Response({'email': 'Successfully activated'}, status=status.HTTP_200_OK)
+        except jwt.ExpiredSignatureError as identifier:
+            return Response({'error': 'Activation Expired'}, status=status.HTTP_400_BAD_REQUEST)
+        except jwt.exceptions.DecodeError as identifier:
+            return Response({'error': 'Invalid token'}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class PasswordRecoveryView(APIView):
+    permission_classes = [AllowAny]
+
+    def post(self, request):
+        email = request.data.get('email')
+
+        serializer = RecoveryEmailSerializer(data=request.data)
+        if serializer.is_valid():
+            try:
+                user = CustomUser.objects.get(email=email)
+                token = RefreshToken.for_user(user).access_token
+                Util.send_recovery_email(get_current_site(request).domain, user, token)
+                return Response({'success': 'Email was sent successfully'}, status=status.HTTP_200_OK)
+            except CustomUser.DoesNotExist:
+                return Response({'error': 'User does not exist'}, status=status.HTTP_400_BAD_REQUEST)
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class PasswordResetView(APIView):
+    permission_classes = [AllowAny]
+
+    def get(self, request):
+        return Response({'success': 'Enter new data'}, status=status.HTTP_200_OK)
+
+    def post(self, request, token):
+        serializer = PasswordResetSerializer(data=request.data)
+        if serializer.is_valid():
+            try:
+                payload = jwt.decode(token, settings.SECRET_KEY, settings.SIMPLE_JWT['ALGORITHM'])
+                user = CustomUser.objects.get(id=payload['user_id'])
+                user.set_password(serializer.validated_data.get('password'))
+                user.save()
+                return Response({'success': 'Password has been successfully updated'}, status=status.HTTP_200_OK)
```

Comment:
You are decoding the token and resetting the password if valid. Consider also verifying the user's session or other context-specific details to enhance security